src/main/java/com/minecolonies/coremod/blocks/BlockInfoPoster.java

```diff
@@ -79,33 +81,40 @@ public IBlockState getStateFromMeta(final int meta)
     @Override
     public int getMetaFromState(final IBlockState state)
     {
-        return ((EnumFacing) state.getValue(FACING)).getIndex();
+        return state.getValue(FACING).getIndex();
     }
 
+    @NotNull
+    @Deprecated
     @Override
-    public IBlockState withRotation(final IBlockState state, final Rotation rot)
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
     {
-        return state.withProperty(FACING, rot.rotate((EnumFacing) state.getValue(FACING)));
+        return state.withProperty(FACING, rot.rotate(state.getValue(FACING)));
     }
 
+    @NotNull
+    @Deprecated
     @Override
-    public IBlockState withMirror(final IBlockState state, final Mirror mirrorIn)
+    public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mirrorIn)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1133)